pkg/apis/eksctl.io/v1alpha3/types.go

```diff
@@ -181,31 +241,48 @@ func (c *ClusterConfig) NewNodeGroup() *NodeGroup {
 // NodeGroup holds all configuration attributes that are
 // specific to a nodegroup
 type NodeGroup struct {
-	Name string
-
-	AMI               string
-	AMIFamily         string
-	InstanceType      string
-	AvailabilityZones []string
-	Tags              map[string]string
-	PrivateNetworking bool
-
-	DesiredCapacity int
-	MinSize         int
-	MaxSize         int
-
-	VolumeSize int
-
-	Labels         NodeLabels
-	MaxPodsPerNode int
-
-	PolicyARNs      []string
-	InstanceRoleARN string
-
-	AllowSSH         bool
-	SSHPublicKeyPath string
-	SSHPublicKey     []byte
-	SSHPublicKeyName string
+	Name string `json:"name"`
+	// +optional
+	AMI string `json:"ami,omitempty"`
+	// +optional
+	AMIFamily string `json:"amiFamily,omitempty"`
+	// +optional
+	InstanceType string `json:"instanceType,omitempty"`
+	// +optional
+	AvailabilityZones []string `json:"availabilityZones,omitempty"`
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+	// +optional
+	PrivateNetworking bool `json:"privateNetworking"`
+
+	DesiredCapacity int `json:"desiredCapacity"`
+	// +optional
+	MinSize int `json:"minSize,omitempty"`
+	// +optional
+	MaxSize int `json:"maxSize,omitempty"`
+
+	// +optional
+	VolumeSize int `json:"volumeSize"`
+	// +optional
+	MaxPodsPerNode int `json:"maxPodsPerNode,omitempty"`
+
+	// +optional
+	Labels NodeLabels `json:"labels,omitempty"`
+
+	// +optional
+	PolicyARNs []string `json:"policyARNs,omitempty"`
+	// +optional
+	InstanceRoleARN string `json:"instanceRoleARN,omitempty"` // TODO: read-only
```

Comment:
I wish this field kept writable.

The use-case is that the user pre-creates role used by nodes to provide proper necessary permissions  including iam policy statements and trust relationship, to make kube2iam/kiam/[aws-iam-controller](mikkeloscar/kube-aws-iam-controller) work.

Reply:
Very good point, but isn't `PolicyARNs` actually sufficient for that?

Actually, the TODO needs to be removed. I've added it initially thinking that there was some kind of Kubernetes tag (like `+optional`) that sets a field as read-only, but after digging-in I failed to find such tag. So TODO is to remove the TODO comment :)

Reply:
Thanks for the response! `PolicyARNs` has potential to partly solve the problem. I've written a more complete description in #398